src/components/cylc/tree/cylc-tree.js

```diff
@@ -420,14 +422,16 @@ class CylcTree {
     if (taskProxy) {
       this.recursivelyRemoveNode(taskProxy)
       // Remember that we attach task proxies children of 'root' directly to a cycle point!
-      let parent
-      if (taskProxy.node.firstParent.id.endsWith('|root')) {
-        const cyclePointId = getCyclePointId(taskProxy)
-        parent = this.lookup.get(cyclePointId)
-      } else {
-        parent = this.lookup.get(taskProxy.node.firstParent.id)
+      if (taskProxy.node.firstParent) {
+        let parent
+        if (taskProxy.node.firstParent.id.endsWith('|root')) {
+          const cyclePointId = getCyclePointId(taskProxy)
+          parent = this.lookup.get(cyclePointId)
+        } else {
+          parent = this.lookup.get(taskProxy.node.firstParent.id)
+        }
+        parent.children.splice(parent.children.indexOf(taskProxy), 1)
```

Comment:
This code block is identical to the one above in the CylcTree class, except for the last line - is it worth sharing to avoid potentially getting them out of sync in future?

Reply:
Excellent idea @hjoliver ! I've added a `findTaskProxyParent` function that should reduce duplicity. Thanks!!!

Reply:
(they were already a bit out of sync, BTW... initially I was doing the `if (taskProxy.node.firstParent.id.endsWith('|root'))` check, until I realized I didn't need to use `.endsWith`, since GraphQL provides the `node.name` attribute with the root family name; well spotted)

Reply:
Sadly, the first excellent idea I've had regards UI coding :rofl: